### ISSUE TITLE
Setting a propertygroup to a tab causes issues during import

### ISSUE THREAD

**Issue by LennardF1989** [umbraco/Umbraco.Deploy.Issues#84]
Deploy 4.3.0 - seems to be mostly related to compositions.

Reproduction is probably:
- Lets say you have a SEO (composition) and Home (doctype). Home has SEO as a composition. 
- If you create a propertygroup with the name "Media" (not tab!) in SEO and add a property to it, then save/deploy that to another environment.
- Change the SEO so the propertygroup becomes a tab (in the UDA, only the Type should change from 0 to 1, nothing else).
- Save/deploy to another environment. During import it will fail.

Exact errors change from time to time, after a few failed imports you even have to recycle the website because it corrupts the cache.

I'll update as I find more, but want to report this that SOMETHING is up with tabs.

**Comment by robgha01** [comment]
I'm also having this problem, we use the api to generate the property group:

If no property group exists do
ContentType.AddPropertyGroup(propertyGroupAlias, propertyGroupName)

then we get the property group
var propertyGroup = ContentType.PropertyGroups[propertyGroupIndex];
propertyGroup.Type = PropertyGroupType.Tab;

we also add a tab parent if desired by doing 
propertyGroup.UpdateParentAlias(parentTabAlias);

after that is done we save it.

for some reason the deploy throws a error when pushing to staging
`Message:
Failed with exception.

									
Exception:
Umbraco.Deploy.Exceptions.ProcessArtifactException: Process pass #2 failed for artifact umb://document-type/2b3a1e01eb2e482ebd5d6612480a769d. ---> System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.ObjectModel.KeyedCollection`2.AddKey(TKey key, TItem item)
   at System.Collections.ObjectModel.KeyedCollection`2.InsertItem(Int32 index, TItem item)
   at Umbraco.Core.Models.PropertyGroupCollection.InsertItem(Int32 index, PropertyGroup item)
   at Umbraco.Core.Models.PropertyGroupCollection.Reset(IEnumerable`1 groups)
   at Umbraco.Deploy.Connectors.ServiceConnectors.ContentTypeConnector.ContentTypeConnectorBase`2.Pass2(DeployState state)
   at Umbraco.Deploy.Connectors.ServiceConnectors.ContentTypeConnector.DocumentTypeConnector.Pass2(DeployState state)
   at Umbraco.Deploy.Connectors.ServiceConnectors.ContentTypeConnector.ContentTypeConnectorBase`2.Process(DeployState state, IDeployContext context, Int32 pass)
   at Umbraco.Deploy.Environments.CurrentEnvironment.ProcessArtifacts(IEnumerable`1 arts, IDeployContext context, CancellationToken token)
   --- End of inner exception stack trace ---
   at Umbraco.Deploy.Environments.CurrentEnvironment.ProcessArtifacts(IEnumerable`1 arts, IDeployContext context, CancellationToken token)
   at Umbraco.Deploy.Environments.CurrentEnvironment.ProcessPackageAsync(Guid sessionId, Package package, CancellationToken token)
   at Umbraco.Deploy.Work.WorkItems.DiskReadWorkItem.<Proceed>d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Umbraco.Deploy.Work.WorkItems.DiskReadWorkItem.<ExecuteAsyncSub>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Umbraco.Deploy.Work.WorkItems.DiskReadWorkItem.<ExecuteAsync>d__15.MoveNext()`

if we remove that uda file from staging and the page from backoffice and clear the database and memory caches then add the uda file back from development into staging data/revisions and run echo > deploy the page is generated with no tabs everything is normal groups and the tab parents is created as regular empty groups. I will try and add a reproduction that you can use.

**Comment by robgha01** [comment]
If you still cant reproduce this bug give me a call and the customer cloud environment "Internationella Engelska Skolan" where this happens is up for tests.

**Comment by AndyButland** [comment]
Thanks for the report.  I believe we've found this issue and have a fix due for Deploy 4.4.0, which is [currently available in RC](https://umbraco.com/blog/umbraco-forms-8891-and-deploy-4491-release-candidates/).  If you'd be happy to try this out Umbraco Support can arrange access for you to update your Cloud project to 4.4.0-rc002.

**Comment by AndyButland** [comment]
Will close and mark this as fixed for 4.4.0, as it does sound like the same issue we've found and resolved from another report.

**Comment by md-hc** [comment]
@AndyButland Unfortunately we are experiencing this issue, even with 4.4.0.

The composition was changed from a property group to a tab. When deployed, it errors.

```
Umbraco.Core.Exceptions.InvalidCompositionException: ContentType with alias 'pageSummary' has an invalid composition 
and there was a conflict on the following PropertyTypes: ''. PropertyTypes must have a unique alias across all 
Compositions in order to compose a valid ContentType Composition.
```

It extracts fine on a clean DB, only fails on the existing database that still has the property group.

Uda file (the only change was the type from 0 to 1, I have highlighted)
```
{
  "Name": "Page Summary",
  "Alias": "pageSummary",
  "DefaultTemplate": null,
  "AllowedTemplates": [],
  "Icon": "icon-document",
  "Thumbnail": "folder.png",
  "Description": null,
  "IsContainer": false,
  "Permissions": {
    "AllowVaryingByCulture": false,
    "AllowVaryingBySegment": false,
    "AllowedAtRoot": false,
    "IsElementType": false,
    "AllowedChildContentTypes": []
  },
  "Parent": "umb://document-type-container/1fab536474a74c0d829600175f7de062",
  "CompositionContentTypes": [],
  "PropertyGroups": [
    {
      "Key": "d45bb30b-6b20-4b90-9af4-77537d436b75",
      "Name": "Summary",
      "SortOrder": 10,
      "Type": 1, // This was changed from 0 to 1
      "Alias": "summary",
      "PropertyTypes": [
        {
          "Key": "274630e4-6756-4e49-8be2-4152f5627393",
          "Alias": "pageSummaryTitle",
          "DataType": "umb://data-type/853bf35887cd464082e5b6f9eedc66ff",
          "Description": "Enter the title that appears on listing, card collection and related pages components. If not set it will use the page name.\n\nIf specified this value will be used when sharing the page on social media.",
          "Mandatory": false,
          "Name": "Title",
          "SortOrder": 2,
          "Validation": "",
          "VariesByCulture": false,
          "VariesBySegment": false,
          "LabelOnTop": false,
          "MemberCanEdit": false,
          "ViewOnProfile": false,
          "IsSensitive": false
        },
        {
          "Key": "3b8d8055-0154-444b-80f6-3e10d63b28ed",
          "Alias": "pageSummaryImage",
          "DataType": "umb://data-type/135d60e064d949edab08893c9ba44ae5",
          "Description": "Select the image that appears on the listing page, card collection or the related pages section of other 'Post' pages. If not set it will use the default image found under website setting",
          "Mandatory": false,
          "Name": "Image",
          "SortOrder": 0,
          "Validation": "",
          "VariesByCulture": false,
          "VariesBySegment": false,
          "LabelOnTop": false,
          "MemberCanEdit": false,
          "ViewOnProfile": false,
          "IsSensitive": false
        },
        {
          "Key": "b0a8cef7-dbf5-4ce5-96ec-e6b09ee0262e",
          "Alias": "pageSummarySubline",
          "DataType": "umb://data-type/13ba557d6c5b48aa9c8823c6cfa273e8",
          "Description": "Enter a brief description about this article. This will appear on the cards in the listing, card collections and related pages components. If not set the meta description will be used.",
          "Mandatory": false,
          "Name": "Subline",
          "SortOrder": 3,
          "Validation": "",
          "VariesByCulture": false,
          "VariesBySegment": false,
          "LabelOnTop": false,
          "MemberCanEdit": false,
          "ViewOnProfile": false,
          "IsSensitive": false
        }
      ]
    }
  ],
  "PropertyTypes": [],
  "Udi": "umb://document-type/2ab2335b002a4c6abb5736954c2e700b",
  "Dependencies": [
    {
      "Udi": "umb://data-type/135d60e064d949edab08893c9ba44ae5",
      "Ordering": true,
      "Mode": 0
    },
    {
      "Udi": "umb://data-type/13ba557d6c5b48aa9c8823c6cfa273e8",
      "Ordering": true,
      "Mode": 0
    },
    {
      "Udi": "umb://data-type/853bf35887cd464082e5b6f9eedc66ff",
      "Ordering": true,
      "Mode": 0
    },
    {
      "Udi": "umb://document-type-container/1fab536474a74c0d829600175f7de062",
      "Ordering": true,
      "Mode": 0
    }
  ],
  "__type": "Umbraco.Deploy,Umbraco.Deploy.Artifacts.ContentType.DocumentTypeArtifact",
  "__version": "4.3.0"
}
```

Full error:

```
previous location where exception was thrown ---\r\n   at 
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at 
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at 
Umbraco.Deploy.Work.WorkItems.DiskReadWorkItem.<ExecuteAsync>d__15.MoveNext()\r\n\r\nINNER:\r\n\tUmbraco.Core.Exception
s.InvalidCompositionException: ContentType with alias 'pageSummary' has an invalid composition and there was a conflict
 on the following PropertyTypes: ''. PropertyTypes must have a unique alias across all Compositions in order to compose
 a valid ContentType Composition.\r\n\t   at 
Umbraco.Core.Services.Implement.ContentTypeServiceBase`3.ValidateLocked(TItem compositionContentType) in 
D:\\a\\1\\s\\src\\Umbraco.Core\\Services\\Implement\\ContentTypeServiceBaseOfTRepositoryTItemTService.cs:line 111\r\n\t
   at Umbraco.Core.Services.Implement.ContentTypeServiceBase`3.Save(TItem item, Int32 userId) in 
D:\\a\\1\\s\\src\\Umbraco.Core\\Services\\Implement\\ContentTypeServiceBaseOfTRepositoryTItemTService.cs:line 423\r\n\t
   at 
Umbraco.Deploy.Connectors.ServiceConnectors.ContentTypeConnector.DocumentTypeConnector.SaveContentType(IContentType 
contentType)\r\n\t   at 
Umbraco.Deploy.Connectors.ServiceConnectors.ContentTypeConnector.ContentTypeConnectorBase`2.Process(DeployState state, 
IDeployContext context, Int32 pass)\r\n\t   at 
Umbraco.Deploy.Connectors.ServiceConnectors.ServiceConnectorBase`3.Process(ArtifactDeployState dart, IDeployContext 
context, Int32 pass)\r\n\t   at Umbraco.Deploy.Environments.CurrentEnvironment.ProcessArtifacts(IEnumerable`1 arts, 
IDeployContext context, CancellationToken 
token)\r\n\r\nEXCEPTION:\r\n\r\nUmbraco.Deploy.Exceptions.ProcessArtifactException: Process pass #2 failed for artifact
 umb://document-type/2ab2335b002a4c6abb5736954c2e700b.\r\n   at 
Umbraco.Deploy.Environments.CurrentEnvironment.ProcessArtifacts(IEnumerable`1 arts, IDeployContext context, 
CancellationToken token)\r\n   at Umbraco.Deploy.Environments.CurrentEnvironment.ProcessPackageAsync(Guid sessionId, 
Package package, CancellationToken token)\r\n   at 
Umbraco.Deploy.Work.WorkItems.DiskReadWorkItem.<Proceed>d__17.MoveNext()\r\n--- End of stack trace from previous 
location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task 
task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at 
Umbraco.Deploy.Work.WorkItems.DiskReadWorkItem.<ExecuteAsyncSub>d__16.MoveNext()\r\n--- End of stack trace from 
previous location where exception was thrown ---\r\n   at 
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at 
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at 
Umbraco.Deploy.Work.WorkItems.DiskReadWorkItem.<ExecuteAsync>d__15.MoveNext()\r\n\r\nINNER:\r\n\tUmbraco.Core.Exception
s.InvalidCompositionException: ContentType with alias 'pageSummary' has an invalid composition and there was a conflict
 on the following PropertyTypes: ''. PropertyTypes must have a unique alias across all Compositions in order to compose
 a valid ContentType Composition.\r\n\t   at 
Umbraco.Core.Services.Implement.ContentTypeServiceBase`3.ValidateLocked(TItem compositionContentType) in 
D:\\a\\1\\s\\src\\Umbraco.Core\\Services\\Implement\\ContentTypeServiceBaseOfTRepositoryTItemTService.cs:line 111\r\n\t
   at Umbraco.Core.Services.Implement.ContentTypeServiceBase`3.Save(TItem item, Int32 userId) in 
D:\\a\\1\\s\\src\\Umbraco.Core\\Services\\Implement\\ContentTypeServiceBaseOfTRepositoryTItemTService.cs:line 423\r\n\t
   at 
Umbraco.Deploy.Connectors.ServiceConnectors.ContentTypeConnector.DocumentTypeConnector.SaveContentType(IContentType 
contentType)\r\n\t   at 
Umbraco.Deploy.Connectors.ServiceConnectors.ContentTypeConnector.ContentTypeConnectorBase`2.Process(DeployState state, 
IDeployContext context, Int32 pass)\r\n\t   at 
Umbraco.Deploy.Connectors.ServiceConnectors.ServiceConnectorBase`3.Process(ArtifactDeployState dart, IDeployContext 
context, Int32 pass)\r\n\t   at Umbraco.Deploy.Environments.CurrentEnvironment.ProcessArtifacts(IEnumerable`1 arts, 
IDeployContext context, CancellationToken token)\r\n\r\n","MismatchList":null}
```

**Comment by LennardF1989** [comment]
@md-hc Your UDA is v4.3.0 though. Can you try again by doing a re-export first? I've seen it once or twice, but I'm pretty sure it was just a fluke. We did notice that if you messed with older versions and broke things thoroughly, you will have to rebuild the whole database from scratch :/

**Comment by md-hc** [comment]
@LennardF1989 Thanks for spotting that - same issue though when version is v4.4.0 (updated the uda file now).

**Comment by md-hc** [comment]
@LennardF1989 @AndyButland Just to update on this, I resolved the issue. It was caused by:

https://github.com/umbraco/Umbraco-CMS/issues/11416

To work around this Umbraco issue, I renamed the property group to "Summary Temp", and then renamed it back to "Summary" which worked great, but caused the Deploy error.

For Deploy, giving the tab a different name to the original property group solved the issue (called it "Page Summary").
